core/src/kura.rs

```diff
@@ -523,11 +509,91 @@ impl BlockStoreTrait for StdFileBlockStore {
         Ok(())
     }
 
-    fn write_block_index(&mut self, block_height: u64, start: u64, length: u64) -> Result<()> {
+    /// Call `read_block_indices` with a buffer of one.
+    ///
+    /// # Errors
+    /// IO Error.
+    pub fn read_block_index(&self, block_height: u64) -> Result<(u64, u64)> {
+        let mut index = (0, 0);
+        self.read_block_indices(block_height, std::slice::from_mut(&mut index))?;
+        Ok(index)
+    }
+
+    /// Get the number of indices in the index file, which is calculated as the size of
+    /// the index file in bytes divided by 16.
+    ///
+    /// # Errors
+    /// IO Error.
+    ///
+    /// The most common reason this function fails is
+    /// that you did not call `create_files_if_they_do_not_exist`.
+    ///
+    /// Note that if there is an error, you can be quite sure all other
+    /// read and write operations will also fail.
+    #[allow(clippy::integer_division)]
+    pub fn read_index_count(&self) -> Result<u64> {
+        let index_file = std::fs::OpenOptions::new()
+            .read(true)
+            .open(self.path_to_blockchain.join(INDEX_FILE_NAME))?;
+        Ok(index_file.metadata()?.len() / (2 * std::mem::size_of::<u64>() as u64))
+        // Each entry is 16 bytes.
+    }
+
+    /// Read block data starting from the `start_location_in_data_file` in data file
+    /// in order to fill `dest_buffer`.
+    ///
+    /// # Errors
+    /// IO Error.
+    pub fn read_block_data(
+        &self,
+        start_location_in_data_file: u64,
+        dest_buffer: &mut [u8],
+    ) -> Result<()> {
+        let mut data_file = std::fs::OpenOptions::new()
+            .read(true)
+            .open(self.path_to_blockchain.join(DATA_FILE_NAME))?;
+        data_file.seek(SeekFrom::Start(start_location_in_data_file))?;
+        data_file.read_exact(dest_buffer)?;
+        Ok(())
+    }
+}
+
+/// Functions available only for lock-protected block store,
+/// i.e. ones writing to it.
```

Comment:
```suggestion
/// i.e. writing.
```

Reply:
Could you elaborate please? Your suggestion feels overly succinct to me.

Reply:
I'd make this similar to lines 438 and 461, maybe this is better:

```suggestion
/// i.e. write operations.
```